scroll-search.lic

```diff
@@ -0,0 +1,533 @@
+		
+	  scroll_information.set_ordinal(new_ordinal)
+	  scroll_information.set_container_index(new_index)
+	end
+  end
+
+  def resave_container(container, scroll_info, remove)
+  
+	# if wasRemoved is true, we remove the scroll from
+	# the container. If not, we simply add it to the container specified.
+	
+	if remove
+	  scrolls_in_container = @owned_scrolls[scroll_info.container]
+	  removed = scrolls_in_container.delete(scroll_info)
+	  if removed
+		echo("Deleting scroll #{removed.spell} from container #{scroll_info.container}") if $debug_mode
+		# Re-save owned scrolls listing
+		@owned_scrolls[scroll_info.container] = scrolls_in_container
+			
+		# Re-save user variables
+		UserVars.owned_scrolls = @owned_scrolls
+	  else
+		echo_error("COULD NOT TRANSFER SCROLL TO NEW CONTAINER")
+		exit
+	  end
+	else
+	  echo ("Attempting to add scroll #{scroll_info.spell} to container #{container}.") if $debug_mode
+		
+	  my_scroll_list = []
+
+	  if @owned_scrolls.key? (container)
+		my_scroll_list = @owned_scrolls[container]
+	  end
+
+	  my_scroll_list.push(scroll_info)
+	  if my_scroll_list.length > 0
+		echo("Adding scroll #{scroll_info.spell} to container #{container}. Length of array is #{my_scroll_list.length}") if $debug_mode
+		# Re-save owned scrolls listing
+		@owned_scrolls[container] = my_scroll_list
+			
+		# Re-save user variables
+		UserVars.owned_scrolls = @owned_scrolls
+			
+		# List scrolls
+		my_scroll_list.each do |scroll_array|
+		  echo("Scrolls in container is #{scroll_array.spell}") if $debug_mode
+		end
+			
+		# List everything
+		@owned_scrolls.each do |container, scroll_array|
+		  scroll_array.each do |scroll_information|
+			echo("Container #{container} | #{scroll_information.spell}") if $debug_mode
+		  end
+		end
+	  else
+		echo_message("COULD NOT ADD SCROLL TO CONTAINER")
+		exit
+	  end
+	end
+  end
+  
+  def grab_scroll(index)
+	echo ("Index in grab scroll of scroll is: #{index}.") if $debug_mode
+	
+    # Retrieve scroll info 
+	scroll_info = get_scroll_info(index)
+	echo("Got scroll info with spell: #{scroll_info.spell}") if $debug_mode
+
+	# Get previous values
+	cur_scroll_container = get_noun(scroll_info.container)
+	cur_scroll_ordinal = scroll_info.ordinal
+	cur_scroll_type = scroll_info.get_type
+
+	bput("open my #{cur_scroll_container}", 'That is already open', 'You open')
+	bput("get #{cur_scroll_ordinal} #{cur_scroll_type} from my #{cur_scroll_container}", 'You get')
+	
+	# Reorder container
+	reorder_container(scroll_info.container, scroll_info, true)
+	
+	# Remove this scroll from our hash map
+	scrolls_in_container = @owned_scrolls[scroll_info.container]
+	
+	echo("Scrolls in container count is: #{scrolls_in_container.length}") if $debug_mode
+	scrolls_in_container.each do |scroll_information|
+	  echo ("Index of current scroll in container is: #{scroll_information.index}. Index is: #{index}") if $debug_mode
+	  if scroll_information.index != index.to_i
```

Comment:
You could do something like `scrolls_in_container.select { |scroll_information| scroll_information.index == index.to_i }.each do ...` to remove the if and else from this block.